src/orion/algo/axoptimizer.py

```diff
@@ -0,0 +1,344 @@
+                    self.register(new_trial)
+                    trials.append(new_trial)
+                    self._trials_map[self.get_id(new_trial)] = trial_index  # tmp
+
+        return trials
+
+    def observe(self, trials):
+        """Observe the `trials` new state of result.
+
+        TODO: document how observe work for this algo
+
+        Parameters
+        ----------
+        trials: list of ``orion.core.worker.trial.Trial``
+           Trials from a `orion.algo.space.Space`.
+
+        """
+        with self.get_client() as ax_client:
+            for trial in trials:
+                if not self.has_suggested(trial):
+                    _, trial_index = ax_client.attach_trial(
+                        AxOptimizer.transform_params(flatten(trial.params), self.space)
+                    )
+                    self._trials_map[self.get_id(trial)] = trial_index
+
+                if not self.has_observed(trial):
+                    # Check the trial status
+                    trial_status = trial.status
+
+                    # If the trial status is `completed`
+                    if trial_status == "completed":
+                        # Complete it in Ax
+                        ax_trial_index = self._trials_map[self.get_id(trial)]
+                        raw_data = {
+                            "objective": trial.objective.value,
+                            **{
+                                s.name: s.value
+                                for s in trial.statistics
+                                if s.name in self.extra_objectives
+                            },
+                            **{r.name: r.value for r in trial.constraints},
+                        }
+                        ax_client.complete_trial(
+                            trial_index=ax_trial_index, raw_data=raw_data
+                        )
+
+                    # If the trial status is `broken`
+                    elif trial_status == "broken":
+                        # Set is as broken is Ax
+                        ax_trial_index = self._trials_map[self.get_id(trial)]
+                        ax_client.log_trial_failure(ax_trial_index)
+
+                    # Register the unobserved trial
+                    self.register(trial)
+
+    @property
+    def is_done(self):
+        """Return True, if an algorithm holds that there can be no further improvement."""
+        # NOTE: Drop if base implementation is fine.
+        return super().is_done
+
+    @property
+    def space(self):
+        """Domain of problem associated with this algorithm's instance."""
+        return super().space
+
+    @space.setter
+    def space(self, space):
+        """Set space."""
+        _space = self._space
+        self._space = space
+
+        if _space != space:
+            # Cleanup and test Ax parameters
+            self._client_state = None
+            with self.get_client():
+                pass
+            self._client_state = None
```

Comment:
```suggestion
```
The part testing `self.get_client` can now be moved to `__init__` thanks to #833.